Engine/Module/ModuleResourceManager.cpp

```diff
@@ -39,136 +79,244 @@ update_status ModuleResourceManager::PreUpdate()
 	return update_status::UPDATE_CONTINUE;
 }
 
- bool ModuleResourceManager::CleanUp()
+bool ModuleResourceManager::CleanUp()
 {
-	 thread_comunication.stop_thread = true;
-	 importing_thread.join();
+	thread_comunication.stop_thread = true;
+	importing_thread.join();
 	return true;
 }
 
  void ModuleResourceManager::StartThread()
  {
-	 bool success = App->filesystem->CreateMountedDir("Library");
-	 bool force_import = !App->filesystem->Exists("Library");
-	 if (force_import && !success)
-	 {
-		 return;
-	 }
-
 	 thread_comunication.finished_loading = false;
-	 thread_comunication.total_items = App->filesystem->assets_file->total_sub_files_number;
-	 ImportAllFilesInDirectory(*App->filesystem->assets_file.get(), force_import);
+	 thread_comunication.total_items = App->filesystem->assets_folder_path->total_sub_files_number;
+
+	 CleanInconsistenciesInDirectory(*App->filesystem->assets_folder_path); // Clean all incorrect meta in the folder Assets.
+	 ImportAssetsInDirectory(*App->filesystem->assets_folder_path); // Import all assets in folder Assets. All metafiles in Assets are correct"
+	// TODO: We need also to iterate over the library folder and to delete binaries that doesnt have corresponding metafiles
 	 thread_comunication.finished_loading = true;
 	 last_imported_time = thread_timer->Read();
  }
 
- ImportResult ModuleResourceManager::Import(const File& file, bool force)
- {
-	 while (thread_comunication.thread_importing_hash == std::hash<std::string>{}(file.file_path))
-	 {
-		 Sleep(1000);
-	 } 
-	 thread_comunication.main_importing_hash = std::hash<std::string>{}(file.file_path);
-	 ImportResult  result = InternalImport(file, force);
-	 thread_comunication.main_importing_hash = 0;
-	 return result;
- }
-
+void ModuleResourceManager::CleanInconsistenciesInDirectory(const Path& directory_path)
+{
+	for (auto & path_child : directory_path.children)
```

Comment:
Instead of doing recursion two times, one for deleting and one for importing. I would say it is better to do both things at the same time. You are duplicating cost here.

Reply:
You are right here. I did it in two separate methods because other way the vector that was being iterated (the one with the object `Path` that were children of the current path) changed size during the loop, causing memory accessing errors. But I just realized that this happens also here when deleting files, so it mus be changed